state.go

```diff
@@ -317,6 +325,38 @@ func (m *Memberlist) probeNode(node *nodeState) {
 	m.suspectNode(&s)
 }
 
+// Ping initiates a ping to the node with the specified name.
+func (m *Memberlist) Ping(node string, addr net.IP, port uint16) (time.Duration, error) {
+	// Prepare a ping message and setup an ack handler.
+	ping := ping{SeqNo: m.nextSeqNo(), Node: node}
+	ackCh := make(chan ackMessage, m.config.IndirectChecks+1)
+	m.setAckChannel(ping.SeqNo, ackCh, m.config.ProbeInterval)
```

Comment:
What happens if we have two concurrent calls to `Ping()`? Since this method is public that might cause some confusing results. Maybe we should wrap this in a mutex, or at least document that it is not allowed.